server/proxy/pf_server.c

```diff
@@ -313,6 +320,17 @@ static DWORD WINAPI pf_server_handle_client(LPVOID arg)
 
 fail:
 
+	if (ps->disp)
+	{
+		if (ps->dispOpened)
```

Comment:
i think `dispOpened` flag is redundant. checking `ps->disp` should be enough. 
we should set it to null on free as well.

Reply:
Nope.
ps disp is initialized in any successful connection, but it is opened only if it is opened by original client too. Therefore, this flag is needed for knowing if we should call Close.

Reply:
ok